src/psyclone/psyGen.py

```diff
@@ -5840,55 +5882,200 @@ def _number_handler(self, node, parent):
+        argument.
+
+        :param access: how the symbol is accessed within the local scope.
+        :type access: :py:class:`psyclone.psyGen.Symbol.Access`
+        '''
+        def __init__(self, access=None):
+            super(Symbol.Argument, self).__init__(access=access)
+            self._pass_by_value = False
+
+        @property
+        def is_argument(self):
+            '''
+            :returns: Whether or not this symbol is a routine argument.
+            :rtype: bool
+            '''
+            return True
+
+        def __str__(self):
+            return "Argument(pass-by-value={0})".format(self._pass_by_value)
+
+    class FortranGlobal(SymbolInterface):
+        '''
+        Describes the interface to a Fortran Symbol representing data that
+        exists outside of the local scope but is not supplied as a routine
+        argument.
+
+        :param str module_use: the name of the Fortran module from which the \
+                               symbol is imported.
+        :param access: the manner in which the Symbol is accessed in the \
+                       associated code section. If None is supplied then the \
+                       access is Symbol.Access.UNKNOWN.
+        :type access: :py:class:`psyclone.psyGen.Symbol.Access` or None.
+        '''
+        def __init__(self, module_use, access=None):
+            self._module_name = ""
+            super(Symbol.FortranGlobal, self).__init__(access=access)
+            self.module_name = module_use
+
+        def __str__(self):
+            return "FortranModule({0})".format(self.module_name)
+
+        @property
+        def module_name(self):
+            '''
+            :returns: the name of the Fortran module from which the symbol is \
+                      imported or None if it is not a module variable.
+            :rtype: str or None
+            '''
+            return self._module_name
+
+        @module_name.setter
+        def module_name(self, value):
+            '''
+            Setter for the name of the Fortran module from which this symbol
+            is imported.
+
+            :param str value: the name of the Fortran module.
+
+            :raises TypeError: if the supplied value is not a str.
+            :raises ValueError: if the supplied string is not at least one \
+                                character long.
+            '''
+            if not isinstance(value, str):
+                raise TypeError("module_name must be a str but got '{0}'".
+                                format(type(value)))
+            if not value:
+                raise ValueError("module_name must be one or more characters "
+                                 "long")
+            self._module_name = value
+
+        @property
+        def is_argument(self):
+            '''
+            :returns: Whether or not this symbol is a routine argument.
+            :rtype: bool
+            '''
+            return False
```

Comment:
Again, if we add more SymbolInterface sub-classes in the future, maybe we want this is_argument->False in the base class and just redefine it at the `Argument` class.

Reply:
I've removed `is_argument` altogether as we only ever use it in the tests :-)